ls1intum/Artemis
# `General`: Add auto-complete mechanism to the user group editor field

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
This PR is related to and closes this [issue](https://github.com/ls1intum/Artemis/issues/6587).

### Description
This PR implements a feature that adds an autocomplete function for the group names. With that, admins can easily add the respective group names to the users preventing non-existent groups from being added.

### Steps for Testing
Prerequisites:
- 1 admin user
- 1 existing non-admin user

1. Log in to Artemis as admin
2. Navigate to Server Administration
3. Choose User Management
4. Select a user and try to add the user to a group
5. Observe adding existing/non-existing groups to the user 
6. Save and make sure that the group is added to the user

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage




| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| CourseResource.java | 94% | ✅                           |
| user-management-update.component.ts | 86.23% | ✅              |
| course-management.service.ts | 87.5% | ✅              |


### Screenshots
<img width="1006" alt="image" src="https://github.com/ls1intum/Artemis/assets/81355607/83772d60-4c6e-402f-bff5-36311b0fc433">
